src/core/ext/filters/client_channel/http_proxy.cc

```diff
@@ -176,7 +187,8 @@ class HttpProxyMapper : public ProxyMapperInterface {
     grpc_arg args_to_add[2];
     args_to_add[0] = grpc_channel_arg_string_create(
         const_cast<char*>(GRPC_ARG_HTTP_CONNECT_SERVER),
-        const_cast<char*>(absl::StripPrefix(uri->path(), "/").data()));
+        const_cast<char*>(
+            MaybeAddDefaultPort(absl::StripPrefix(uri->path(), "/")).data()));
```

Comment:
Please use `c_str()` instead of `data()` here.  (Pre-existing bug.)

Reply:
starting c++ 11 c_str() and data() do the same thing https://en.cppreference.com/w/cpp/string/basic_string/data

Reply:
made the change anyway